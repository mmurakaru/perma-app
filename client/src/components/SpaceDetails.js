import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { v4 as uuidv4 } from 'uuid'
import { ReactComponent as ArrowIcon } from '../assets/arrow_down.svg'
import { ReactComponent as Sunflower } from '../assets/sunflower.svg'

SpaceDetails.propTypes = {
    space: PropTypes.object,
    switchToPreviousPage: PropTypes.func,
}

export default function SpaceDetails({ space, switchToPreviousPage }) {
    let today = new Date()
    const dd = String(today.getDate()).padStart(2, '0')
    const mm = String(today.getMonth() + 1).padStart(2, '0') //January is 0!
    const yyyy = today.getFullYear()
    today = dd + '.' + mm + '.' + yyyy

    return (
        <BackgroundStyled>
            <ArrowDown data-testid="ArrowDown" onClick={switchToPreviousPage} />
            <DetailsContainer>
                <SpaceWrapper>
                    <SpaceIcon>
                        <LogoStyled />
                    </SpaceIcon>
                    <h2>{space.spaceTitle}</h2>
                </SpaceWrapper>
                <PlantsWrapper>
                    {space.plants.map((plant) => (
                        <Plant key={uuidv4()}>
                            <PlantImage src={plant.image_url} alt="" />
                            <PlantInfo>
                                <h2>{plant.common_name}</h2>
                                <span>{today}</span>
                            </PlantInfo>
                        </Plant>
                    ))}
                </PlantsWrapper>
            </DetailsContainer>
        </BackgroundStyled>
    )
}

const PlantsWrapper = styled.div`
    margin: 60px 0 60px;
    overflow: auto;
`

const Plant = styled.div`
    border: 0.2px solid var(--medium-grey);
    border-radius: 6px;
    padding: 5px 0 5px;
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin-top: 5px;
`

const PlantInfo = styled.div`
    margin-left: 20px;
    display: grid;
    grid-template: column;
    gap: 5px;

    h2 {
        font-weight: 400;
        font-size: 0.8rem;
    }

    span {
        font-size: 0.6rem;
        font-weight: 200;
    }
`

const PlantImage = styled.img`
    width: 40px;
    height: 40px;
    border-radius: 50%;
    margin-left: 10px;
`

const BackgroundStyled = styled.div`
    background: var(--medium-green);
    width: 100%;
    height: 100%;
    padding-top: 150px;
`

const ArrowDown = styled(ArrowIcon)`
    fill: var(--white);
    width: 20px;
    position: absolute;
    left: 20px;
    top: 20px;
`

const DetailsContainer = styled.section`
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    padding: 20px;
    background-color: var(--white);
    width: 100%;
    overflow: none;
    position: relative;
    height: 100%;
`

const SpaceIcon = styled.div`
    height: 80px;
    width: 80px;
    border-radius: 5px;
    display: grid;
    place-items: center;
    border: none;
    background-color: var(--light-green);
`
const LogoStyled = styled(Sunflower)`
    width: 80%;
`

const SpaceWrapper = styled.div`
    display: flex;
    flex-direction: column;
    position: absolute;
    top: -50px;
    h2 {
        margin: 3px 0;
        opacity: 50%;
    }
`
