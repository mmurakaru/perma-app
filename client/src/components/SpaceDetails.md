SpaceDetails example:

```js
<SpaceDetails
    space={{
        title: 'Vegetables',
        plants: [
            {
                image_url:
                    'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                common_name: 'Coconut',
            },
            {
                image_url:
                    'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                common_name: 'Strawberry',
            },
        ],
    }}
/>
```
