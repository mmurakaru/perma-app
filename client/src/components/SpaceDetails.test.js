import SpaceDetails from './SpaceDetails'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('SpaceDetails', () => {
    it('renders correctly', () => {
        const { container } = render(
            <SpaceDetails
                plant={{
                    image_url:
                        'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                }}
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    it('calls onClick', () => {
        const onClickMock = jest.fn()
        const { getByTestId } = render(
            <SpaceDetails
                switchToSpaceOverview={onClickMock}
                plant={{
                    image_url:
                        'https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed',
                }}
            />
        )
        userEvent.click(getByTestId('ArrowDown'), 'Coconut')
        expect(onClickMock).toHaveBeenCalled()
    })
})
