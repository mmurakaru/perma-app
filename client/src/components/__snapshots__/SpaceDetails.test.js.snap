// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceDetails renders correctly 1`] = `
.c6 {
  margin-top: 60px;
  height: 80px;
}

.c7 {
  border: 0.2px solid var(--mediumGrey);
  border-radius: 6px;
  padding: 5px 0 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c9 {
  margin-left: 20px;
  display: grid;
  grid-template: column;
  gap: 5px;
}

.c9 h2 {
  font-weight: 400;
  font-size: 0.8rem;
}

.c9 span {
  font-size: 0.6rem;
  font-weight: 200;
}

.c8 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  margin-left: 10px;
}

.c0 {
  background: var(--mediumGreen);
  width: 100%;
  height: 100%;
  padding-top: 150px;
}

.c1 {
  fill: var(--white);
  width: 20px;
  position: absolute;
  left: 20px;
  top: 20px;
}

.c2 {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  box-shadow: rgba(0,0,0,0.24) 0px 3px 8px;
  padding: 20px;
  background-color: var(--white);
  width: 100%;
  overflow: none;
  position: relative;
  height: 100%;
}

.c4 {
  height: 80px;
  width: 80px;
  border-radius: 5px;
  display: grid;
  place-items: center;
  border: none;
  background-color: var(--lightGreen);
}

.c5 {
  width: 80%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: -50px;
}

.c3 h2 {
  margin: 3px 0;
  opacity: 50%;
}

<div
  class="c0"
>
  <svg
    class="c1"
    data-testid="ArrowDown"
  >
    arrow_down.svg
  </svg>
  <section
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <svg
          class="c5"
        >
          banana.svg
        </svg>
      </div>
      <h2 />
    </div>
    <div
      class="c6"
    >
      <div
        class="c7"
      >
        <img
          alt=""
          class="c8"
          src="https://bs.floristic.org/image/o/63073d2fbf45b90701279405ecc2eec0272906ed"
        />
        <div
          class="c9"
        >
          <h2 />
          <span>
            02.12.2020
          </span>
        </div>
      </div>
    </div>
  </section>
</div>
`;
